lib/AST/Type.cpp

```diff
@@ -3221,6 +3223,24 @@ Type TypeBase::getSuperclassForDecl(const ClassDecl *baseClass) {
   llvm_unreachable("no inheritance relationship between given classes");
 }
 
+Type TypeBase::getGenericAncestor() {
+  Type t = getConcreteTypeForSuperclassTraversing(this);
+
+  while (t) {
+    auto NTD = t->getAnyNominal();
+    assert(NTD && "expected nominal type in NTD");
```

Comment:
Should we be tolerant of error types here, e.g., `assert(NTD || t->hasError())`?

Reply:
I changed the loop to `while (t && !t->hasError())`.